surefire-its/src/test/resources/testng-parallel-suites/src/test/java/testng/suiteXml/TestNGSuiteTest.java

```diff
@@ -28,16 +28,23 @@
  * @author <a href="mailto:tibordigana@apache.org">Tibor Digana (tibor17)</a>
  * @since 2.19
  */
-public class TestNGSuiteTest {
-	private static final AtomicInteger counter = new AtomicInteger();
+public class TestNGSuiteTest
+{
+    private static final AtomicInteger counter = new AtomicInteger();
 
-	@Test
-	public void shouldRunAndPrintItself()
-		throws Exception
-	{
-		System.out.println( getClass().getSimpleName() + "#shouldRunAndPrintItself() "
-								+ counter.incrementAndGet() + ".");
+    @Test
+    public void shouldRunAndPrintItself() throws Exception
+    {
+        String message = String.format(
+                "%s#shouldRunAndPrintItself() %d.",
+                getClass().getSimpleName(),
+                counter.incrementAndGet()
+        );
+        TimeUnit.SECONDS.sleep( 1 );
 
-		TimeUnit.SECONDS.sleep( 2 );
-	}
+        synchronized ( System.out )
```

Comment:
I've explained it in commit message: https://github.com/apache/maven-surefire/commit/2beb507001867dc61f2be55d206c5825e9f0c2ca

> Fixing flaky test Surefire1177TestngParallelSuitesIT
> 
> This test rely on displaying messages on System.out, but invocations wasn't synchronized so messeges tend to interweave and test wasn't able to assert for them properly. I added synchronization and move code a litle bit.

Reply:
@cardil 
Thx for pointing out the synchronization issue with `System.out`.
I have noticed the same problem yesterday, see https://github.com/cbeust/testng/issues/1908
Treating the stream in test would fix our builds but we still have this issue on Ubuntu and not on Windows. Does it mean that the Sysout should be synchronized in the application run? We are setting the delegate via `System.setOut()` and the delegate in `ForkingRunListener` is synchronized. Why we do not see same issue in JUnit tests?

Reply:
It depends on how you write to `OutputStream`. Usually you got a single writer, like with using a logging framework. Every thread gets the same instance of writer. Such writer will usually buffer writing to a stream and in that way, each mesage is treated as separate. 

In this test, there are seperate threads that write bytes directly to `OutputStream`, without buffering or synchronization. This is the reason that here synchronization helps. But it might or might not be needed in production code. It depends.